containers-contrib-readonly/src/Map/Contrib/Bench.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Map.Contrib.Bench where
+
+import Map
+import Gauge.Main (Benchmark, bench, defaultMain, whnf)
+import Relude hiding (fromList, Map)
+
+benchmark
+  :: forall k. (Enum k, Bounded k, Key k, NFData (Map k Int))
+  => String
+  -> Proxy k
+  -> IO ()
+benchmark label _ = do
+  let benchMaxElems = min 1024 (fromEnum (maxBound @k))
+  let upBound = foldl' (.) id (replicate benchMaxElems succ) minBound
+  let map_keys :: [k] = [minBound..upBound]
```

Comment:
The complexity here exists because I wanted to be able to write benchmarks for any `k` such that `(Enum k, Bounded k, Key k, NFData (Map k Int))` holds. I think it's a good choice but the choice is arguable, obviously.

Reply:
Extra complexity is okay if it allows to have polymoprhic implementation. I see lookup benchmarks from the `haskell-perf/dictionaries` and I'm horrified by their code...

https://github.com/haskell-perf/dictionaries/blob/2bbc950e9597eea74130a714662b9b2c173ecd02/Time.hs#L307-L321

So I think that you're doing great job here 👍